OpenUpSA/wazimap-ng-ui
# boundary type selection box

## Description
Introducing a dropdown to switch between alternate geography types such as mainplace and ward and populating it with the available children boundary types


## Related Issue
https://trello.com/c/wN9UGlrj/454-we-need-to-reintroduce-switching-between-alternate-geographies-such-as-mainplace-and-ward-previously-this-was-hardcoded-it-shoul

## How to test it locally
* Navigate to a geography that has children with more than 1 boundary types. Example : /#geo:CPT on beta.youthexplorer.org.za instance. Instance can be changed by hardcoding hostname variable in index.js as 
`let hostname = 'beta.youthexplorer.org.za';`
* Boundary type selection box can be seen on bottom right side of the page
* Change boundary type and see that the current geograph is redrawn accordingly

## Screenshots
![image](https://user-images.githubusercontent.com/53019884/93765844-34023600-fc1e-11ea-8cfd-5ce4873c479f.png)
![image](https://user-images.githubusercontent.com/53019884/93765854-36fd2680-fc1e-11ea-8804-534aeaab75b6.png)


## Changelog

### Added
* Added `js/map/boundary_type_box.js` to manage boundary type selection box functions
* Added `js/setup/boundaryevents.js` to manage boundary type selection box related events

### Updated
* Added necessary references of `js/map/boundary_type_box.js` and `js/setup/boundaryevents.js` to `js/load.js`
* Updated `js/controller.js` to trigger `reDraw()` function whenever an option is selected from the boundary type selection dropdown
* Updated `js/setup/map.js` for `overlayBoundaries()` function to be called whenever `reDraw()` is triggered

### Removed


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
